tasks.py

```diff
@@ -0,0 +1,117 @@
+import os
+from pathlib import Path
+from shutil import which
+
+from invoke import task
+from invoke.util import cd
+
+TOOLS = ["poetry", "pre-commit"]
+PORT = os.environ.get("SERVER_PORT", 8000)
+
+ACTIVE_VENV = os.environ.get("VIRTUAL_ENV", None)
+VENV_HOME = Path(os.environ.get("WORKON_HOME", "~/.local/share/virtualenvs"))
+VENV_PATH = Path(ACTIVE_VENV) if ACTIVE_VENV else (VENV_HOME / "kagi")
+VENV = str(VENV_PATH.expanduser())
+
+PRECOMMIT = which("pre-commit") if which("pre-commit") else (VENV / "pre-commit")
+POETRY = which("poetry") if which("poetry") else (VENV / "poetry")
+
+
+@task
+def build(c, docs=True):
+    """Build documentation"""
+    if docs:
+        c.run(f"{VENV}/bin/sphinx-build docs docs/_build")
+
+
+@task(build)
+def livedocs(c):
+    """Serve docs at http://localhost:$PORT/ (default port is 8000)"""
+    from livereload import Server
+
+    server = Server()
+    server.watch("docs/conf.py", lambda: build(c))
+    server.watch("CONTRIBUTING.rst", lambda: build(c))
+    server.watch("docs/*.rst", lambda: build(c))
+    server.serve(port=PORT, root="docs/_build")
```

Comment:
Should we have a `DOC_PORT` and a `DEMO_PORT`?

Reply:
I thought about that. I wasn't sure whether anyone would want to have both running at the same time or not. But I suppose we can separate the options and let users decide. :smile:

Reply:
Separate `DOCS_PORT` and `DEMO_PORT` variables now exist as of bad1205.